deploy/sdk/src/dynamo/sdk/lib/decorators.py

```diff
@@ -90,8 +92,15 @@ def decorator(func: t.Callable) -> DynamoEndpoint:
     return decorator
 
 
-def async_on_start(func: t.Callable) -> t.Callable:
+def async_on_start(func: F) -> F:
     """Decorator for async onstart functions."""
     # Mark the function as a startup hook
     setattr(func, "__dynamo_startup_hook__", True)
     return func
+
+
+def async_on_shutdown(func: F) -> F:
+    """Decorator for async onshutdown functions."""
+    # Mark the function as a shutdown hook
+    setattr(func, "__dynamo_shutdown_hook__", True)
```

Comment:
I searched the codebase, and I can't find anything that triggers this hook's logic. Is it missing?


For example, the `async_on_start` is implemented in the `dynamo/deploy/sdk/src/dynamo/sdk/cli/serve_dynamo.py`

```python
    @dynamo_worker()
    async def dyn_worker(runtime: DistributedRuntime):
    ...
            # Run startup hooks before setting up endpoints
            for name, member in vars(class_instance.__class__).items():
                if callable(member) and getattr(
                    member, "__dynamo_startup_hook__", False
                ):
```